test/e2e/session_test.go

```diff
@@ -3,201 +3,123 @@ package e2e_test
 import (
 	"fmt"
 	"math/rand"
-	"os"
+	"regexp"
+	"strings"
 	"time"
 
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	y "github.com/onsi/ginkgo"
+	z "github.com/onsi/gomega"
 
-	"github.com/solo-io/solo-kit/pkg/api/v1/resources/core"
-	v1 "github.com/solo-io/squash/pkg/api/v1"
-	sqOpts "github.com/solo-io/squash/pkg/options"
-	squashcli "github.com/solo-io/squash/pkg/squashctl"
+	gokubeutils "github.com/solo-io/go-utils/kubeutils"
 	"github.com/solo-io/squash/test/testutils"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
 )
 
-//const KubeEndpoint = "http://localhost:8001/api"
-
 func Must(err error) {
-	Expect(err).NotTo(HaveOccurred())
+	z.Expect(err).NotTo(z.HaveOccurred())
+}
+func check(err error) {
+	z.Expect(err).NotTo(z.HaveOccurred())
 }
 
-var (
-	daName  = "debug-attachment-1"
-	daName2 = "debug-attachment-2"
-	// testNamespace = "squash-debugger-test2"
-	testNamespace = "stest"
-	testNSRoot    = "stest"
-	testsStarted  = 0
-)
-
-var _ = Describe("Single debug mode", func() {
+var _ = y.Describe("Single debug mode", func() {
+	testutils.DeclareTestConditions()
 
 	seed := time.Now().UnixNano()
 	fmt.Printf("rand seed: %v\n", seed)
 	rand.Seed(seed)
 
-	var (
-		params testutils.E2eParams
-	)
-
-	// Deploy the services that you will debug
-	BeforeEach(func() {
-		testsStarted++
-		// Use unique namespaces so we can start tests before namespace is deleted
-		// Use predictable namespaces so that we can establish watches
-		// (solo-kit does not have a "watch all namespaces" feature yet)
-		if os.Getenv("SERIALIZE_NAMESPACES") != "1" {
-			testNamespace = fmt.Sprintf("%v-%v", testNSRoot, rand.Int31n(100000))
-		} else {
-			testNamespace = fmt.Sprintf("%v-%v", testNSRoot, testsStarted)
-		}
-		params = testutils.NewE2eParams(testNamespace, daName, GinkgoWriter)
-		params.SetupE2e()
+	y.It("Should create a debug session", func() {
+		cs := &kubernetes.Clientset{}
+		y.By("should get a kube client")
+		restCfg, err := gokubeutils.GetConfig("", "")
+		check(err)
+		cs, err = kubernetes.NewForConfig(restCfg)
+		check(err)
+
+		y.By("should list no resources after delete")
+		err = testutils.Squashctl("utils delete-attachments")
+		check(err)
+		str, err := testutils.SquashctlOut("utils list-attachments")
+		check(err)
+		validateUtilsListDebugAttachments(str, 0)
+
+		// create namespace
+		testNamespace := fmt.Sprintf("testsquash-%v", rand.Intn(1000))
+		y.By("should create a demo namespace")
+		_, err = cs.CoreV1().Namespaces().Create(&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testNamespace}})
+		check(err)
+
+		y.By("should deploy a demo app")
+		err = testutils.Squashctl(fmt.Sprintf("deploy demo --demo-id %v --demo-namespace1 %v --demo-namespace2 %v", "go-java", testNamespace, testNamespace))
+		check(err)
+		time.Sleep(5 * time.Second)
```

Comment:
are you wait for the pod to be in ready state?
i suggest explicitly testing that in a loop, in case it takes more than 5 seconds every now and then.